Test/Controller/BidSpec.hs

```diff
@@ -1,20 +1,23 @@
-module Web.Controller.BidSpec where
+module Test.Controller.BidSpec where
 
 import Network.HTTP.Types.Status (status200)
 
-import IHP.FrameworkConfig (ConfigBuilder (..))
 import IHP.Prelude
 import IHP.QueryBuilder (query)
 import IHP.Test.Mocking
+import IHP.Fetch
+
+import IHP.FrameworkConfig
+import IHP.HaskellSupport
 import Test.Hspec
+import Config
 
 import Generated.Types
-import Main ()
 import Web.Routes
 import Web.Types
 
 spec :: Spec
-spec = beforeAll (makeConfig >>= mockContext WebApplication) do
+spec = beforeAll (mockContext WebApplication config) do
```

Comment:
@mpscholten calling you here, hope it's ok - so I could adapt test docs accordingly :smile_cat: 

With this change we now get

```
 :l Test/Main
[1 of 6] Compiling Config           ( Config/Config.hs, interpreted )
[2 of 6] Compiling Generated.Types  ( build/Generated/Types.hs, interpreted )
[3 of 6] Compiling Web.Types        ( Web/Types.hs, interpreted )
[4 of 6] Compiling Web.Routes       ( Web/Routes.hs, interpreted )
[5 of 6] Compiling Test.Controller.BidSpec ( Test/Controller/BidSpec.hs, interpreted )

Test/Controller/BidSpec.hs:20:19: error:
    * No instance for (IHP.ControllerSupport.InitControllerContext
                         WebApplication)
        arising from a use of `mockContext'
    * In the first argument of `beforeAll', namely
        `(mockContext WebApplication config)'
      In the expression:
        beforeAll
          (mockContext WebApplication config)
          do describe "User controller"
               $ do it "has no existing users" $ withContext do ...
                    it "responds with some content" $ withContext do ...
                    ....
      In an equation for `spec':
          spec
            = beforeAll
                (mockContext WebApplication config)
                do describe "User controller"
                     $ do it "has no existing users" $ withContext ...
                          ....
   |
20 | spec = beforeAll (mockContext WebApplication config) do
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Test/Controller/BidSpec.hs:27:24: error:
    * Could not deduce (IHP.ControllerSupport.Controller
                          UsersController)
        arising from a use of `mockActionResponse'
      from the context: ContextParameters WebApplication
        bound by a type expected by the context:
                   ContextParameters WebApplication => IO ()
        at Test/Controller/BidSpec.hs:(26,55)-(28,58)
    * In a stmt of a 'do' block:
        content <- mockActionResponse NewUserAction
      In the first argument of `withContext', namely
        `do content <- mockActionResponse NewUserAction
            content `shouldBe` "<!DOCTYPE HTML>...</html>"'
      In the second argument of `($)', namely
        `withContext
           do content <- mockActionResponse NewUserAction
              content `shouldBe` "<!DOCTYPE HTML>...</html>"'
   |
27 |             content <- mockActionResponse NewUserAction
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Test/Controller/BidSpec.hs:31:13: error:
    * Could not deduce (IHP.ControllerSupport.Controller
                          UsersController)
        arising from a use of `mockActionStatus'
      from the context: ContextParameters WebApplication
        bound by a type expected by the context:
                   ContextParameters WebApplication => IO ()
        at Test/Controller/BidSpec.hs:(30,79)-(31,70)
    * In the first argument of `shouldReturn', namely
        `mockActionStatus CreateUserAction'
      In a stmt of a 'do' block:
        mockActionStatus CreateUserAction `shouldReturn` status200
      In the second argument of `withParams', namely
        `do mockActionStatus CreateUserAction `shouldReturn` status200'
   |
31 |             mockActionStatus CreateUserAction `shouldReturn` status200
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Test/Controller/BidSpec.hs:34:28: error:
    * Could not deduce (IHP.ControllerSupport.Controller
                          UsersController)
        arising from a use of `mockAction'
      from the context: ContextParameters WebApplication
        bound by a type expected by the context:
                   ContextParameters WebApplication => IO ()
        at Test/Controller/BidSpec.hs:(33,54)-(35,54)
    * In the first argument of `headers', namely
        `(mockAction NewUserAction)'
      In a stmt of a 'do' block: hs <- headers (mockAction NewUserAction)
      In the first argument of `withContext', namely
        `do hs <- headers (mockAction NewUserAction)
            lookup "Location" hs `shouldNotBe` Nothing'
   |
34 |             hs <- headers (mockAction NewUserAction)
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^
Failed, four modules loaded.
```

Reply:
This looks like it's an actual bug in your app :D 

Can you call https://ihp.digitallyinduced.com/api-docs/IHP-Assets-ControllerFunctions.html#v:initAssetVersion `initAssetVersion` from `    initContext = do` inside your `Web/FrontController`?

Reply:
I'ved pushed a possible fix for the `initAssetVersion` problem to https://github.com/digitallyinduced/ihp/pull/1141 Can you give that branch a try? It should work without putting the `initAssetVersion` into the `Web/FrontController.hs`